%\section{Further perspectives} Partie Ã  retravailler en perspectives et conclusion

%We have designed here a project that intends to draw theoretical evolutionary predictions from first principles and to foster the dialog between functional and evolutionary approaches in order to better understand why organisms work the way they do. Because further developments on both ends of the project are contingent to their results, it is not possible to anticipate precisely the next steps, but we have already put forward how some of these natural steps will emerge from the project. Two other concomitant perspectives deserve to be considered at this stage.

%First, in the present outline of the project, the environment is considered to be constant such that fitness landscapes always look exactly the same, an unreasonable assumption for \textit{in vivo} conditions. Acknowledging that ecological conditions impact Adaptation dates all the way back to the very beginning of Evolutionary biology insofar as it is the tenet of Natural Selection identified by \cite{Darwin59} and \cite{Wallace58}: different environments can be seen as different fitness landscapes \citep{Wright31} where the optima are not located at the same place. Conversely, this means that species evolving under different environment are not subject to the same selective pressure, which therefore contribute to define evolutionary outcomes such as the mutation-selection-drift balance. Complexity is indeed ubiquitous in Nature \citep{Bergelson21}, even in simplified systems \citep{Sanchez-Gorostiaga19}. This is all the more true since environment are always subject to fluctuations - albeit with different magnitude, temporality and stochasticity - a phenomenon which has largely been proven to determine optimal phenotypic strategies such as predictive plasticity \citep{Gotthard95,Ghalambor07}, bet-hedging \citep{Cohen66,Slatkin74,Cooper82,Philippi89} or polymorphism \citep{Wittmann17}, be it among macro- \citep{Menu93,Philippi93,Harpak21} and micro-organisms \citep{Ratcliff10,Solopova14} or in between \citep{Martinez-Garcia17}.\citet{Wilke01b} and \citet{Mustonen08} compellingly unraveled how it should impact the long-term selective coefficient $s$, but it was not until recently that these ecological pressures were shown to impair severely the strength of Natural Selection under certain circumstances \citep{Cvijovic15}. Besides, it was also put forward that fitness landscapes may turn out to be deformable such that the power of selection would be even more contingent to the evolutionary and ecological history \citep{Bajic18}. Introducing ecological factors and other phenomenon likely to harm the potential of Natural Selection \citep{Graves17} cannot thus be overlooked when one is willing to make accurate predictions about \textit{in vivo} Evolution and should be added for further developments on either part of this project.

%Simultaneously, it could also become relevant to attempt to detect complementary epistasis in genomic data using well identified and characterized genes and pathways (\textit{eg.} genes involved in glycolysis) among closely related species - sharing similar ecological niches - or even within populations, whose genetic divergence concerns few loci. Identifying genes and/or loci of the same pathway that face distinct selective pressures depending on the genomic background to which they belong may yield testable predictions about the actual metabolic effects of mutations even when they depart from \textit{a priori} (for example, a mutation improving catalytic properties of an enzyme is supposed to increase the metabolic flux and may well do so - or not - in different organisms but still have few or no effect at all on the fitness in some but not all of them, because in those where it does not, it is not the fitness limiting step) and to reconcile expectations about fitness effects of mutations with their actual counterpart. Reciprocally, investigating why some mutations increasing fluxes\footnote{In principle, it could also be done with organs for instance, or symbiotic relationships or any complementary interaction, but the link with fitness for a seemingly improved parameter is yet more complex and makes it all but attainable at this stage.} display similar fitness effects than synonymous mutations would help appreciate how existing phenotypes and their underlying genotypes are translated into fitness while it should also feeds the community with information about mechanisms responsible for adaptation in functional sites such as the catalytic site of an enzyme.

%Then, using species with more or less different life histories and phylogenetic relatedness, it may become practicable to test these ideas on a wider scale as some pathways are largely conserved, and to see whether it influences the inference of population features (such as $N_e$). Obviously, there are many contributions that can equally account for similar selective signatures and the objective being to disentangle these manifold contributions, data analysis needs either be restricted to typical cases where flux, fitness and genomic index of adaptations can readily be determined, or to be seen from a broader perspective. This suggests another long-term line of research, where the combination of such a framework with the knowledge of other mechanisms involved in molecular fitness - namely protein stability \citep{Dasmeh14,Dasmeh18,Echave17a,Echave19} - would aim at explaining part of the variability observed when characterizing the fitness of specific sites or codons along and across phylogenies \citep{Rodrigue10,Rodrigue17,Parto17,Jones20} and may help to better grasp adaptation signals as was shown in \citep{Dasmeh14}, and also to identify what governs the adaptive process, a yet contentious issue \citep{Harpak21}. Along with many other lines of research, for instance on convergent Adaptation \citep{Stoltzfus17}, such an effort would also help uncover which part of the evolutionary process is contingent and which one is necessary \citep{Monod71,Ben-Menahem97}, for a similar strength of selection may yield pathways whose weakest links are spread differently, and, eventually, lead to variability in levels of evolvability.